downloader/src/main/kotlin/vcs/Mercurial.kt

```diff
@@ -71,9 +71,9 @@ object Mercurial : VersionControlSystem() {
         // We cannot detect beforehand if the Large Files extension would be required, so enable it by default.
         val extensionsList = mutableListOf(EXTENSION_LARGE_FILES)
 
-        if (vcsRevision != null) {
+        if (!vcsRevision.isNullOrEmpty()) {
```

Comment:
Better extend the original code by `&& vcsRevision.isNotBlank()` to still be able to benefit from the smart cast, and to check for blank strings, too, to be on the safe side.